metaflow/main_cli.py

```diff
@@ -324,6 +332,51 @@ def show(profile):
     else:
         echo('Configuration is set to run locally.')
 
+@configure.command(help='Exports the configuration to a file.')
+@click.option('--profile', '-p', default='',
+              help="Optional user profile whose configuration must be "
+                   "exported.")
+@click.argument('output_filename', type=click.Path(resolve_path=True))
+def export(profile, output_filename):
+    check_for_missing_profile(profile)
+    # Export its contents to a new file.
+    path = get_config_path(profile)
+    env_dict = {}
+    with open(path, 'r') as f:
+        env_dict = json.load(f)
+    # resolve_path doesn't expand `~` in `path`.
+    output_path = expanduser(output_filename)
+    if os.path.exists(output_path):
+        if click.confirm('Do you wish to overwrite the contents in ' +
+                         click.style('"%s"' % output_path, fg='cyan') + '?',
+                         abort=True):
+            pass
+    # Write to file.
+    with open(output_path, 'w') as f:
+        json.dump(env_dict, f, indent=4, sort_keys=True)
+    echo('Configuration successfully exported to: ', nl=False)
+    echo('"%s"' % output_path, fg='cyan')
+
+@configure.command(help='Imports the configuration from a file.', name='import')
+@click.option('--profile', '-p', default='',
+              help="Optional user profile to which the configuration must be "
+                   "imported into.")
+@click.argument('input_filename', type=click.Path(exists=True,
+                                                  resolve_path=True))
+def import_from(profile, input_filename):
+    check_for_missing_profile(profile)
+    # Import configuration.
+    input_path = expanduser(input_filename)
+    env_dict = {}
+    with open(input_path, 'r') as f:
+        env_dict = json.load(f)
```

Comment:
Maybe catch the exception if the file doesn't contain a valid JSON blob and throw it nicely?

Reply:
From offline discussion with @romain-intel the consensus was to leave this particular case to the default handling provided the python (library) itself, since most often it might be doing the right thing, instead of us massaging the error and re-throwing it.

For this specific case, when I `import` an invalid JSON, I see:

```
<stack trace>
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 3 column 1 (char 18)
```

I'm inclined to leave it as-is unless you insist.

For posterity, the stack trace consists of:
```
Traceback (most recent call last):
  File "/usr/local/bin/metaflow", line 11, in <module>
    load_entry_point('metaflow', 'console_scripts', 'metaflow')()
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
File "/Users/rchirravuri/metaflow_oss/metaflow/main_cli.py", line 373, in import_from
    env_dict = json.load(f)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 296, in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/__init__.py", line 348, in loads
    return _default_decoder.decode(s)
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/Cellar/python/3.7.4_1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/json/decoder.py", line 353, in raw_decode
    obj, end = self.scan_once(s, idx)
```